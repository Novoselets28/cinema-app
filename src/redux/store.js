import { createStore, applyMiddleware, combineReducers } from 'redux';
import createSagaMiddleware from 'redux-saga';
import sessionTimeReducer from './reducers/reducersSessionTime';
import sessionListReducer from './reducers/reducersSessionList';
import sessionTimeSaga from './sagas/sagasSessionTime';
import sessionListSaga from './sagas/sagasSessionList';
import movieSessionsReducer from './reducers/reducersMovieSessions';
import movieSessionsSaga from './sagas/sagasMovieSessions';
import mainPageReducer from './reducers/reducersMainPage';
import mainPageSaga from './sagas/sagasMainPage';
import filmDescriptionReducer from './reducers/filmDescriptionReducer';
import filmDescriptionSaga from './sagas/filmDescriptionSaga';
import cinemaReducer from './reducers/reducersCinema';
import cinemaSaga from './sagas/cinemaSaga';

const sagaMiddleware = createSagaMiddleware();

const rootReducer = combineReducers({
  mainPage: mainPageReducer,
  sessionTime: sessionTimeReducer,
  sessionList: sessionListReducer,
  movieSessions: movieSessionsReducer,
  filmDescription: filmDescriptionReducer,
  cinema: cinemaReducer,

});

const store = createStore(
  rootReducer,
  applyMiddleware(sagaMiddleware)
);

sagaMiddleware.run(mainPageSaga);
sagaMiddleware.run(sessionTimeSaga);
sagaMiddleware.run(sessionListSaga);
sagaMiddleware.run(movieSessionsSaga);
sagaMiddleware.run(filmDescriptionSaga);
sagaMiddleware.run(cinemaSaga);

export default store;
