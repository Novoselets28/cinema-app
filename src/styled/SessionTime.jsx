import { Button } from 'react-bootstrap';
import styled from 'styled-components';

export const StyledButton = styled(Button)`
    
    width: 5rem;
    margin: 1rem;
`;