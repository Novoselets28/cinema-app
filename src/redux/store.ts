import { createStore, applyMiddleware, combineReducers } from 'redux';
import createSagaMiddleware from 'redux-saga';
import sessionTimeReducer, { sessionTimeSaga } from './ducks/sessionTime';
import filmDescriptionReducer, { filmDescriptionSaga } from './ducks/filmDescription';
import cinemaReducer, { cinemaSaga } from './ducks/cinema';
import sessionListReducer, { sessionListSaga } from './ducks/sessionList';
import movieSessionsReducer, { movieSessionsSaga } from './ducks/movieSessions';
import mainPageReducer, { mainPageSaga } from './ducks/mainPage';

export type RootState = ReturnType<typeof rootReducer>;

const sagaMiddleware = createSagaMiddleware();

export const rootReducer = combineReducers({
  mainPage: mainPageReducer,
  sessionTime: sessionTimeReducer,
  sessionList: sessionListReducer,
  movieSessions: movieSessionsReducer,
  filmDescription: filmDescriptionReducer,
  cinema: cinemaReducer,
});

const store = createStore(
  rootReducer,
  applyMiddleware(sagaMiddleware)
);

sagaMiddleware.run(mainPageSaga);
sagaMiddleware.run(sessionTimeSaga);
sagaMiddleware.run(sessionListSaga);
sagaMiddleware.run(movieSessionsSaga);
sagaMiddleware.run(filmDescriptionSaga);
sagaMiddleware.run(cinemaSaga);

export default store;