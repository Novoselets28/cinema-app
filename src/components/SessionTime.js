import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Button } from 'react-bootstrap';
import { useNavigate, useParams } from 'react-router';
import 'bootstrap/dist/css/bootstrap.min.css';
import { StyledButton, StyledContainer } from '../styled/SessionTime';
import { setSelectedSession } from '../redux/actions/actionsSessionTime';

// const SessionTime = () => {
//   const { date } = useParams();
//   const sessions = useSelector((state) => state.sessions);
//   const selectedSession = useSelector((state) => state.selectedSession);
//   const dispatch = useDispatch();
//   const navigate = useNavigate();

//   useEffect(() => {
//     dispatch({ type: 'FETCH_SESSIONS' });
//   }, [dispatch]);

//   const handleTimeSelect = (session) => {
//     dispatch(setSelectedSession(session));
//   };

//   const handleBookClick = () => {
//     if (selectedSession) {
//       navigate(`/cinema/${date}/${selectedSession}`);
//     }
//   };

//   return (
//     <StyledContainer>
//       {sessions.map((session, index) => (
//         <StyledButton
//           key={index}
//           className="transparent-button"
//           onClick={() => handleTimeSelect(session)}
//         >
//           {session}
//         </StyledButton>
//       ))}
//       {selectedSession && (
//         <Button variant="success" onClick={handleBookClick}>
//           Book
//         </Button>
//       )}
//     </StyledContainer>
//   );
// };

const SessionTime = () => {
  const { date } = useParams();
  const sessions = useSelector((state) => state.sessionTime.sessions);
  const selectedSession = useSelector((state) => state.sessionTime.selectedSession);
  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch({ type: 'FETCH_SESSIONS' });
  }, [dispatch]);

  if (sessions.length === 0) {
    return <div>Loading sessions...</div>;
  }

  const handleTimeSelect = (session) => {
        dispatch(setSelectedSession(session));
      };
    
      const handleBookClick = () => {
        if (selectedSession) {
          navigate(`/cinema/${date}/${selectedSession}`);
        }
      };

  return (
    <StyledContainer>
      {sessions.map((session, index) => (
        <StyledButton
          key={index}
          className="transparent-button"
          onClick={() => handleTimeSelect(session)}
        >
          {session}
        </StyledButton>
      ))}
      {selectedSession && (
        <Button variant="success" onClick={handleBookClick}>
          Book
        </Button>
      )}
    </StyledContainer>
  );
};


export default SessionTime;
