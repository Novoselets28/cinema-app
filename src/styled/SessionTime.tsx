import { Button, Container } from 'react-bootstrap';
import styled from 'styled-components';

export const StyledButton = styled(Button)`
    width: 4rem;
    margin: 4px;
`;

export const StyledContainer = styled(Container)`
    padding-top: 2rem;
    padding-bottom: 2rem;
    text-align: center;
`;