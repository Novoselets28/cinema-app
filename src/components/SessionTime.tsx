import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Button } from 'react-bootstrap';
import { useNavigate, useParams } from 'react-router-dom';

import 'bootstrap/dist/css/bootstrap.min.css';

import { fetchSessions, setSelectedSession } from '../redux/ducks/sessionTime';
import { StyledButton, StyledContainer } from '../styled/SessionTime';

interface SessionTimeProps {
  date: string;
}

const SessionTime: React.FC<SessionTimeProps> = () => {
  const { date } = useParams<{ date: string }>();
  const sessions: string[] = useSelector((state: any) => state.sessionTime.sessions);
  const selectedSession: string[] = useSelector((state: any) => state.sessionTime.selectedSession);
  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(fetchSessions());
  }, [dispatch]);

  if (sessions.length === 0) {
    return <div>Loading sessions...</div>;
  }

  const handleTimeSelect = (session: string) => {
    dispatch(setSelectedSession(session));
  };

  const handleBookClick = () => {
    if (selectedSession) {
      navigate(`/cinema/${date}/${selectedSession}`);
    }
  };

  return (
    <StyledContainer>
      {sessions.map((session, index) => (
        <StyledButton
          key={index}
          className="transparent-button"
          onClick={() => handleTimeSelect(session)}
        >
          {session}
        </StyledButton>
      ))}
      {selectedSession && (
        <Button variant="success" onClick={handleBookClick}>
          Book
        </Button>
      )}
    </StyledContainer>
  );
};

export default SessionTime;