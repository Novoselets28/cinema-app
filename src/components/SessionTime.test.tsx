import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/store';

import SessionTime from './SessionTime';

test('renders loading message when sessions are empty', () => {
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/']}>
        <SessionTime date={''} />
      </MemoryRouter>
    </Provider>
  );
  expect(container).toMatchSnapshot();
});

test('renders session buttons when sessions are available', () => {
  const sessions = ['Session 1', 'Session 2', 'Session 3'];

  store.dispatch({ type: 'SET_SESSIONS', payload: sessions });

  const { container } = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/']}>
        <SessionTime date={''} />
      </MemoryRouter>
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
