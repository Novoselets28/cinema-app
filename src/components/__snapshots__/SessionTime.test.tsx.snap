// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading message when sessions are empty 1`] = `
<div>
  <div>
    Loading sessions...
  </div>
</div>
`;

exports[`renders session buttons when sessions are available 1`] = `
<div>
  <div
    class="sc-gEvEer gyOvpF container"
  >
    <button
      class="sc-aXZVg kesiHi transparent-button btn btn-primary"
      type="button"
    >
      Session 1
    </button>
    <button
      class="sc-aXZVg kesiHi transparent-button btn btn-primary"
      type="button"
    >
      Session 2
    </button>
    <button
      class="sc-aXZVg kesiHi transparent-button btn btn-primary"
      type="button"
    >
      Session 3
    </button>
  </div>
</div>
`;
